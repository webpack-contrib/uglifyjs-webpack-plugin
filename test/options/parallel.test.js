import webpack from '../helpers/compiler';
import UglifyJsPlugin from '../../src/index';

const filter = (file) => (!/\.map/.test(file) ? file : false);

describe('Options', () => {
  describe('parallel', () => {
    test('{Boolean} - false', async () => {
      const config = {
        plugins: [
          new UglifyJsPlugin({
            parallel: false,
          }),
        ],
      };

      const stats = await webpack('parallel/entry.js', config);
      const { assets } = stats.compilation;

      Object.keys(assets)
        .filter(filter)
        .forEach((asset) => {
          expect(assets[asset].source()).toMatchSnapshot();
        });
    });

    test('{Boolean} - true', async () => {
      const config = {
        plugins: [
          new UglifyJsPlugin({
            parallel: true,
          }),
        ],
      };

      const stats = await webpack('parallel/entry.js', config);
      const { assets } = stats.compilation;

      Object.keys(assets)
        .filter(filter)
        .forEach((asset) => {
          expect(assets[asset].source()).toMatchSnapshot();
        });
    });

    test('{Number}', async () => {
      const config = {
        plugins: [
          new UglifyJsPlugin({
            parallel: 2,
          }),
        ],
      };

      const stats = await webpack('parallel/entry.js', config);
      const { assets } = stats.compilation;

      Object.keys(assets)
        .filter(filter)
        .forEach((asset) => {
          expect(assets[asset].source()).toMatchSnapshot();
        });
    });
  });
});
