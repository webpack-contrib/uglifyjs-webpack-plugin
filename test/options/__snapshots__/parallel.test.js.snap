// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options parallel {Boolean} - false 1`] = `"(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,o,t){\\"use strict\\";t.r(o),t.d(o,\\"foo\\",function(){return u}),t.d(o,\\"foo2\\",function(){return r}),t.d(o,\\"foo3\\",function(){return f}),t.d(o,\\"foo4\\",function(){return c});function u(n){return n}function r(n){return n}function f(n){return n}function c(n){return baz}}],[[0,1,0]]]);"`;

exports[`Options parallel {Boolean} - false 2`] = `"!function(e){function r(r){for(var t,i,c,l=r[0],s=r[1],p=r[2],f=0,d=[];f<l.length;f++)i=l[f],n[i]&&d.push(n[i][0]),n[i]=0;for(t in s)Object.prototype.hasOwnProperty.call(s,t)&&(e[t]=s[t]);for(a&&a(r);d.length;)d.shift()();for(o.push.apply(o,p||[]),f=0;f<o.length;f++){for(var h=o[f],v=!0,g=1;g<h.length;g++){var m=h[g];0!==n[m]&&(v=!1)}v&&(o.splice(f--,1),c=u(u.s=h[0]))}return c}var t={},n={1:0},o=[];function u(r){if(t[r])return t[r].exports;var n=t[r]={i:r,l:!1,exports:{}};return e[r].call(n.exports,n,n.exports,u),n.l=!0,n.exports}u.e=function(e){var r=[],t=n[e];if(0!==t)if(t)r.push(t[2]);else{var o=new Promise(function(r,o){t=n[e]=[r,o]});r.push(t[2]=o);var i=document.getElementsByTagName(\\"head\\")[0],a=document.createElement(\\"script\\");a.charset=\\"utf-8\\",a.timeout=12e4,u.nc&&a.setAttribute(\\"nonce\\",u.nc),a.src=u.p+\\"\\"+e+\\".bundle.js\\";var c=setTimeout(function(){l({type:\\"timeout\\",target:a})},12e4);a.onerror=a.onload=l;function l(r){a.onerror=a.onload=null,clearTimeout(c);var t=n[e];if(0!==t){if(t){var o=r&&(\\"load\\"===r.type?\\"missing\\":r.type),u=r&&r.target&&r.target.src,i=new Error(\\"Loading chunk \\"+e+\\" failed.\\\\n(\\"+o+\\": \\"+u+\\")\\");i.type=o,i.request=u,t[1](i)}n[e]=void 0}}i.appendChild(a)}return Promise.all(r)},u.m=e,u.c=t,u.d=function(e,r,t){u.o(e,r)||Object.defineProperty(e,r,{configurable:!1,enumerable:!0,get:t})},u.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},u.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return u.d(r,\\"a\\",r),r},u.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},u.p=\\"\\",u.oe=function(e){throw console.error(e),e};var i=window.webpackJsonp=window.webpackJsonp||[],a=i.push.bind(i);i.push=r,i=i.slice();for(var c=0;c<i.length;c++)r(i[c])}([]);"`;

exports[`Options parallel {Boolean} - true 1`] = `"(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,o,t){\\"use strict\\";t.r(o),t.d(o,\\"foo\\",function(){return u}),t.d(o,\\"foo2\\",function(){return r}),t.d(o,\\"foo3\\",function(){return f}),t.d(o,\\"foo4\\",function(){return c});function u(n){return n}function r(n){return n}function f(n){return n}function c(n){return baz}}],[[0,1,0]]]);"`;

exports[`Options parallel {Boolean} - true 2`] = `"!function(e){function r(r){for(var t,i,c,l=r[0],s=r[1],p=r[2],f=0,d=[];f<l.length;f++)i=l[f],n[i]&&d.push(n[i][0]),n[i]=0;for(t in s)Object.prototype.hasOwnProperty.call(s,t)&&(e[t]=s[t]);for(a&&a(r);d.length;)d.shift()();for(o.push.apply(o,p||[]),f=0;f<o.length;f++){for(var h=o[f],v=!0,g=1;g<h.length;g++){var m=h[g];0!==n[m]&&(v=!1)}v&&(o.splice(f--,1),c=u(u.s=h[0]))}return c}var t={},n={1:0},o=[];function u(r){if(t[r])return t[r].exports;var n=t[r]={i:r,l:!1,exports:{}};return e[r].call(n.exports,n,n.exports,u),n.l=!0,n.exports}u.e=function(e){var r=[],t=n[e];if(0!==t)if(t)r.push(t[2]);else{var o=new Promise(function(r,o){t=n[e]=[r,o]});r.push(t[2]=o);var i=document.getElementsByTagName(\\"head\\")[0],a=document.createElement(\\"script\\");a.charset=\\"utf-8\\",a.timeout=12e4,u.nc&&a.setAttribute(\\"nonce\\",u.nc),a.src=u.p+\\"\\"+e+\\".bundle.js\\";var c=setTimeout(function(){l({type:\\"timeout\\",target:a})},12e4);a.onerror=a.onload=l;function l(r){a.onerror=a.onload=null,clearTimeout(c);var t=n[e];if(0!==t){if(t){var o=r&&(\\"load\\"===r.type?\\"missing\\":r.type),u=r&&r.target&&r.target.src,i=new Error(\\"Loading chunk \\"+e+\\" failed.\\\\n(\\"+o+\\": \\"+u+\\")\\");i.type=o,i.request=u,t[1](i)}n[e]=void 0}}i.appendChild(a)}return Promise.all(r)},u.m=e,u.c=t,u.d=function(e,r,t){u.o(e,r)||Object.defineProperty(e,r,{configurable:!1,enumerable:!0,get:t})},u.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},u.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return u.d(r,\\"a\\",r),r},u.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},u.p=\\"\\",u.oe=function(e){throw console.error(e),e};var i=window.webpackJsonp=window.webpackJsonp||[],a=i.push.bind(i);i.push=r,i=i.slice();for(var c=0;c<i.length;c++)r(i[c])}([]);"`;

exports[`Options parallel {Number} 1`] = `"(window.webpackJsonp=window.webpackJsonp||[]).push([[0],[function(n,o,t){\\"use strict\\";t.r(o),t.d(o,\\"foo\\",function(){return u}),t.d(o,\\"foo2\\",function(){return r}),t.d(o,\\"foo3\\",function(){return f}),t.d(o,\\"foo4\\",function(){return c});function u(n){return n}function r(n){return n}function f(n){return n}function c(n){return baz}}],[[0,1,0]]]);"`;

exports[`Options parallel {Number} 2`] = `"!function(e){function r(r){for(var t,i,c,l=r[0],s=r[1],p=r[2],f=0,d=[];f<l.length;f++)i=l[f],n[i]&&d.push(n[i][0]),n[i]=0;for(t in s)Object.prototype.hasOwnProperty.call(s,t)&&(e[t]=s[t]);for(a&&a(r);d.length;)d.shift()();for(o.push.apply(o,p||[]),f=0;f<o.length;f++){for(var h=o[f],v=!0,g=1;g<h.length;g++){var m=h[g];0!==n[m]&&(v=!1)}v&&(o.splice(f--,1),c=u(u.s=h[0]))}return c}var t={},n={1:0},o=[];function u(r){if(t[r])return t[r].exports;var n=t[r]={i:r,l:!1,exports:{}};return e[r].call(n.exports,n,n.exports,u),n.l=!0,n.exports}u.e=function(e){var r=[],t=n[e];if(0!==t)if(t)r.push(t[2]);else{var o=new Promise(function(r,o){t=n[e]=[r,o]});r.push(t[2]=o);var i=document.getElementsByTagName(\\"head\\")[0],a=document.createElement(\\"script\\");a.charset=\\"utf-8\\",a.timeout=12e4,u.nc&&a.setAttribute(\\"nonce\\",u.nc),a.src=u.p+\\"\\"+e+\\".bundle.js\\";var c=setTimeout(function(){l({type:\\"timeout\\",target:a})},12e4);a.onerror=a.onload=l;function l(r){a.onerror=a.onload=null,clearTimeout(c);var t=n[e];if(0!==t){if(t){var o=r&&(\\"load\\"===r.type?\\"missing\\":r.type),u=r&&r.target&&r.target.src,i=new Error(\\"Loading chunk \\"+e+\\" failed.\\\\n(\\"+o+\\": \\"+u+\\")\\");i.type=o,i.request=u,t[1](i)}n[e]=void 0}}i.appendChild(a)}return Promise.all(r)},u.m=e,u.c=t,u.d=function(e,r,t){u.o(e,r)||Object.defineProperty(e,r,{configurable:!1,enumerable:!0,get:t})},u.r=function(e){Object.defineProperty(e,\\"__esModule\\",{value:!0})},u.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return u.d(r,\\"a\\",r),r},u.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},u.p=\\"\\",u.oe=function(e){throw console.error(e),e};var i=window.webpackJsonp=window.webpackJsonp||[],a=i.push.bind(i);i.push=r,i=i.slice();for(var c=0;c<i.length;c++)r(i[c])}([]);"`;
